bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -2169,6 +2335,17 @@ private static void execTypeConversionOpcodes(WorkerExecutionContext ctx, Worker
                 j = operands[1];
                 sf.intRegs[j] = Boolean.parseBoolean(sf.stringRegs[i]) ? 1 : 0;
                 break;
+            case InstructionCodes.S2D:
+                i = operands[0];
+                j = operands[1];
+
+                str = sf.stringRegs[i];
+                try {
+                    sf.refRegs[j] = new BDecimal(new BigDecimal(str, MathContext.DECIMAL128));
```

Comment:
shall we change the BDecimal class so that it's constructor will accept a string and internally it creates the BigDecimal object? along with that you can have relevant methods in BDecimal class as well like "compareTo" etc

Reply:
Created an issue to track this (https://github.com/ballerina-platform/ballerina-lang/issues/11384).